File: generated_tests/csharp/tests/resource_bundles_tests.cs
using Xunit;
using SplashKitSDK;
using static SplashKitSDK.SplashKit;
using HttpMethod = SplashKitSDK.HttpMethod;
namespace SplashKitTests
{
    public class TestResourceBundles
    {
        public TestResourceBundles()
        {
            SetResourcesPath("/mnt/c/Users/Noahc/Documents/aYear_2_semester_2/TeamProject/GitHubRepo/splashkit_test_generator/resources");
        }
        [Fact]
        public void TestFreeResourceBundleIntegration() {
            LoadResourceBundle("Test Resource Bundle", "test.txt");
            using var cleanupResource = new ResourceCleanup("Test Resource Bundle");
            Assert.True(HasResourceBundle("Test Resource Bundle"));
            Assert.True(HasBitmap("FrogBmp"));
            Assert.True(HasFont("hara"));
            FreeResourceBundle("Test Resource Bundle");
            Assert.False(HasResourceBundle("Test Resource Bundle"));
        }
        [Fact]
        public void TestHasResourceBundleIntegration() {
            LoadResourceBundle("Test Resource Bundle", "test.txt");
            using var cleanupResource = new ResourceCleanup("Test Resource Bundle");
            Assert.True(HasResourceBundle("Test Resource Bundle"));
            FreeResourceBundle("Test Resource Bundle");
            Assert.False(HasResourceBundle("Test Resource Bundle"));
            Assert.False(HasResourceBundle("nonexistent"));
        }
        [Fact]
        public void TestLoadResourceBundleIntegration() {
            LoadResourceBundle("Test Resource Bundle", "test.txt");
            using var cleanupResource = new ResourceCleanup("Test Resource Bundle");
            Assert.True(HasResourceBundle("Test Resource Bundle"));
            Assert.True(HasAnimationScript("WalkingScript"));
            Assert.True(HasBitmap("FrogBmp"));
            Assert.True(HasFont("hara"));
            Assert.True(HasSoundEffect("error"));
            Assert.True(HasTimer("my timer"));
            Assert.True(HasResourceBundle("blah"));
            FreeResourceBundle("Test Resource Bundle");
            Assert.False(HasResourceBundle("test_bundle"));
        }
    }
}

File: generated_tests/pascal/tests/resource_bundles_tests.pas
uses SplashKit, TestFramework, ../Helpers;
type
    TTestResourceBundles = class(TTestCase)
    protected
        procedure Setup; override;
    end;
begin
    inherited;
    SetResourcesPath('/mnt/c/Users/Noahc/Documents/aYear_2_semester_2/TeamProject/GitHubRepo/splashkit_test_generator/resources');
end;
procedure TestFreeResourceBundleIntegration;
begin
    LoadResourceBundle('Test Resource Bundle', 'test.txt');
    CleanupResource := TResourceCleanup.Create('Test Resource Bundle')
    AssertTrue(HasResourceBundle('Test Resource Bundle'));
    AssertTrue(HasBitmap('FrogBmp'));
    AssertTrue(HasFont('hara'));
    FreeResourceBundle('Test Resource Bundle');
    AssertFalse(HasResourceBundle('Test Resource Bundle'));
end;
procedure TestHasResourceBundleIntegration;
begin
    LoadResourceBundle('Test Resource Bundle', 'test.txt');
    CleanupResource := TResourceCleanup.Create('Test Resource Bundle')
    AssertTrue(HasResourceBundle('Test Resource Bundle'));
    FreeResourceBundle('Test Resource Bundle');
    AssertFalse(HasResourceBundle('Test Resource Bundle'));
    AssertFalse(HasResourceBundle('nonexistent'));
end;
procedure TestLoadResourceBundleIntegration;
begin
    LoadResourceBundle('Test Resource Bundle', 'test.txt');
    CleanupResource := TResourceCleanup.Create('Test Resource Bundle')
    AssertTrue(HasResourceBundle('Test Resource Bundle'));
    AssertTrue(HasAnimationScript('WalkingScript'));
    AssertTrue(HasBitmap('FrogBmp'));
    AssertTrue(HasFont('hara'));
    AssertTrue(HasSoundEffect('error'));
    AssertTrue(HasTimer('my timer'));
    AssertTrue(HasResourceBundle('blah'));
    FreeResourceBundle('Test Resource Bundle');
    AssertFalse(HasResourceBundle('test_bundle'));
end;
end;

procedure RegisterTests;
begin
    #<Proc:0x00007faa116e2450 /mnt/c/Users/Noahc/Documents/aYear_2_semester_2/TeamProject/GitHubRepo/splashkit_test_generator/test_generator/config/languages/pascal_config.rb:138 (lambda)>
end;
